pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -494,12 +494,10 @@ func doInstallVolumeRef(ctx *zedmanagerContext, config types.AppInstanceConfig,
 	status *types.AppInstanceStatus, vrs *types.VolumeRefStatus) bool {
 
 	changed := false
-	var vrc *types.VolumeRefConfig
-	for i := range config.VolumeRefConfigList {
-		vrc = &config.VolumeRefConfigList[i]
-		if vrc.VolumeID == vrs.VolumeID {
-			break
-		}
+	vrc := getVolumeRefConfigFromAIConfig(&config, *vrs)
```

Comment:
Thanks for fixing that. Could you please move this chunk to a previous commit, where you've introduced this for loop? Better not to fix commits in following commits in the same PR, because someone can get confused reviewing your PR commit by commit (for example me :)

Reply:
of course, sorry for misplacing it, I moved it now

Reply:
@europaul from what I can tell doInstallVolumeRef() is called during a purge and at that point in time we might? have some VolumeRefStatus without a corresponding VolumeRefConfig. I'm not sure. 
In any case, I can't quite understand why you need to add this check.

Reply:
there could be a case when EVE got a config, where VRC has already been deleted but haven't yet deleted the corresponding VRS. anyway it's good to check for a nil pointer here, because we de-reference it in the [next line](https://github.com/lf-edge/eve/blob/e928ad765459cd9471b3cde9ec3077de2d9bb139/pkg/pillar/cmd/zedmanager/updatestatus.go#L506)

Reply:
Sure, but you are adding loging using info from the vrc in this PR.

How can we tell whether e.g., the update of a version of a virtual disk from the controller cause a delete of the old VRC, and whether your added check that the VRC should exist would cause some failure to do that update?

FWIW I ran your PR through the ztests which do such update and purge from the controller and those tests pass. But I'm still concerned that we might be missing something.

Reply:
I think the logic is like this: controller tells EVE what apps are connected to what volumes through VRC's in the AppConfig - this set of VRC's stays constant until the next config from the controller arrives - if a VRC doesn't exist in AppConfig, it also doesn't make sense to publish it through `MaybeAddVolumeRefConfig` to the volumemgr - so the call to `doInstallVolumeRef` can be terminated if a VRC doesn't exist and we can move on to the next one

Reply:
OK.